homeassistant/components/yale_smart_alarm/lock.py

```diff
@@ -0,0 +1,127 @@
+"""Component for interacting with the Yale Smart Alarm System API."""
+import logging
+
+import voluptuous as vol
+from yalesmartalarmclient.client import (
+    AuthenticationError,
+    YaleSmartAlarmClient,
+)
+
+from homeassistant.components.lock import (
+    PLATFORM_SCHEMA,
+    LockEntity,
+)
+from homeassistant.const import (
+    ATTR_CODE,
+    STATE_LOCKED,
+    STATE_UNLOCKED,
+    STATE_UNAVAILABLE,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    CONF_CODE,
+)
+
+import homeassistant.helpers.config_validation as cv
+
+CONF_AREA_ID = "area_id"
+
+DEFAULT_NAME = "Yale Smart Alarm"
+
+DEFAULT_AREA_ID = "1"
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_AREA_ID, default=DEFAULT_AREA_ID): cv.string,
+        vol.Optional(CONF_CODE): cv.string,
+    }
+)
```

Comment:
We no longer allow integrations to add platform YAML configuration.

More information on this can be found in Architecture Decision Record:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>

Please note that this integration connects to a device or service and another Architecture Decision Record applies that disallows the use of YAML configuration in favor of a configuration flow via the UI:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

See our developer documentation on how to get started creating a configuration flow for this integration:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

As these changes often involve a bit of work and some significant shift in the current code, we will close this PR for now.

Thanks already! :+1:

Reply:
I had a hunch this was coming my way.
Ok, will get to it.